pkg/datagatherer/k8s/cache_test.go

```diff
@@ -136,3 +139,14 @@ func TestOnAddCache(t *testing.T) {
 		})
 	}
 }
+
+// TestNoneCache demonstrates that the cache helpers do not crash if passed a
+// non-cachable object, but log an error with a reference to the object type.
```

Comment:
**Nit:** It says that "it logs an error" but we don't assert that something is actually logged. Without seeing the actual log lines, it's hard to see what the test does.

I don't know if it was worth it, but I've opened a PR on top of yours to improve that: #619.